### ISSUE TITLE
hi! can I become your test-guy?

### ISSUE THREAD

**Issue by mickrose555** [newenclave/AtomAppGN#1]
which Garmin model can I get?￼
I know BLE scanner iOS can get the atomfast reading every second in raw format.

is your beta able to keep the reading for 10min ?

I can donate. thanks

**Comment by newenclave** [comment]
Sure. Why not. 
I've not yet published the application to apps.garmin, but I can built it for your device. For now it's really basic, but at least it shows values. 
About devices I'm not sure. The feature with saving "last device" works only on devices with SDK 3.2. 
But if you already have a device, lets try to build the app for it.

**Comment by mickrose555** [comment]
cool! 
i don't have a garmin watch, only an iWatch4. i can get a Garmin watch with BLE ( compatible with app like Goprocontrol...)
any one would be ok? any os?
what matters is stable connection (20min) and update as fast as Android app.

**Comment by newenclave** [comment]
Ok. Lets see. 
This application is for Connect IQ SDK. Here you can find a list with the compatible devices: https://developer.garmin.com/connect-iq/compatible-devices/ (see 3.0 for "Version Compatibility")
There are many of them. I have "Forerunner 945" as am a runner. 
The best "long-battery-play" device now, as far as I know, is "fenix 6" or even "fenix 6 solar edition" but so expensive it is. It's also huge and heavy, but many of people like it. 

BTW: as far as this application is the very first my application for Garmin, I don't know a lot of things about it for now. So I can't answer all the questions that can appear.

**Comment by newenclave** [comment]
@mickrose555 I've updated the Readme. Not all devices can run the application. There must be "Generic Bluetooth Low Energy Channel" feature supported by device.

**Comment by mickrose555** [comment]
ok. i can get a fenix 5X saphire for 270 Eur, almost new.
it's in your URL compatibility list (v3.0)

**Comment by newenclave** [comment]
@mickrose555 yes. fenix 5 is supported. There is an issue with saving "Last used device", but not a big deal for now.

**Comment by newenclave** [comment]
@mickrose555 Looks like "fenix 5x tactix® Charlie" doesn't support BLE :(
Only "fenix 5x plus" supports.

**Comment by mickrose555** [comment]
> @mickrose555 Looks like "fenix 5x tactix® Charlie" doesn't support BLE :(
> 
> Only "fenix 5x plus" supports.
OK! so i need a double list of compatibility sdk 3.0 + BLE

**Comment by mickrose555** [comment]
I can get a forerunner 935 for 180 Eur, almost new. Compatible?

**Comment by newenclave** [comment]
hi @mickrose555  
No, unfortunately is doesn't support "Generic Bluetooth Low Energy Channel" :(

**Comment by zebra108** [comment]
Hello. Ready to become a tester) Device: Vivoactive 4. Do you need to compile the application yourself? is it very difficult for a simple user? or you yourself can compile for a certain model ..?

**Comment by newenclave** [comment]
@zebra108  Hello. 
Yesterday I pusher the very first version of app to the store. But it's a beta now and I have no idea if it's possible to share the beta. 
Vivoactive 4 looks supported. I'll add it to the "supported list" and can build it for you. 

UPD: Yes. It's supported.
![image](https://user-images.githubusercontent.com/5781353/103154657-76995880-47a1-11eb-8b5f-5d92dcb85bed.png)

**Comment by zebra108** [comment]
Great news! Until I found Garmin on the library of applications, perhaps somewhere in the settings you need to enable consent to download beta versions. I will look now.
upd. Could you give me a link in the Garmin Application Library?

**Comment by newenclave** [comment]
@zebra108  Yes. It's here 
https://apps.garmin.com/en-US/apps/d293d486-3da3-4a4a-96de-55cfb6a1c723 
But it's not approved yet. So it's available for no one but me. 

If you have an AtomFast device I can build the PRG file for you.  
Need a place where it can be stored.

**Comment by zebra108** [comment]
Yes, indeed the application is not yet available. I have an Atomfast 8850 device.

upd. Do you want to contact us via telegram, or whatever is convenient for you ..

**Comment by newenclave** [comment]
@zebra108  Ok Here I created a repo https://github.com/newenclave/AtomGNBuilds/tree/master/AtomAppGN/Vivoactive4
You can download the PRG file
Connect your Garmin to PC
Open Garmin/Apps
Copy the file there 
Disconnect you Vivoactive
After that the application should be available in application list. 
I'm available in Telegram as well @blackfailer

**Comment by mickrose555** [comment]
> hi @mickrose555  
> 
> No, unfortunately is doesn't support "Generic Bluetooth Low Energy Channel" :(

hi! 935 is listed as compatible on the new Garmin page!

**Comment by newenclave** [comment]
@mickrose555 hm. Interesting
But it should not.
Probably accidently it was included. 

Yeh. So annoying this Garmin build system is. It KNOWS what a device  supports, but lets it build. 
I'll fix it in the next release.

**Comment by mickrose555** [comment]
i haven't got a watch yet; how complicated for you to build a Fenix 6 release?

**Comment by newenclave** [comment]
@mickrose555 Fenix 6 is already there. But if you are talking about beta to test. Well it's about 3 seconds :)

**Comment by mickrose555** [comment]
> @zebra108  Ok Here I created a repo https://github.com/newenclave/AtomGNBuilds/tree/master/AtomAppGN/Vivoactive4
> 
> You can download the PRG file
> 
> Connect your Garmin to PC
> 
> Open Garmin/Apps
> 
> Copy the file there 
> 
> Disconnect you Vivoactive
> 
> After that the application should be available in application list. 
> 
> I'm available in Telegram as well @blackfailer 

hi! i saw the youtube video. I see atomfast value is small size at the bottom. Can you make that better on a Fenix 6?

**Comment by newenclave** [comment]
@mickrose555  which value?

**Comment by mickrose555** [comment]
> @mickrose555  which value? 

the reading, uSv/h... it's not full size view like your forerunner...

**Comment by newenclave** [comment]
![image](https://user-images.githubusercontent.com/5781353/103480811-2ba1c580-4ddf-11eb-99dc-0368de183041.png)
This?

**Comment by mickrose555** [comment]
yes, the youtube demo by alek, with a vivoactive doesn't look like that.

**Comment by newenclave** [comment]
Vivoactive? Which video are you talking about? I don't have VA :)

**Comment by mickrose555** [comment]
> Vivoactive? Which video are you talking about? I don't have VA :) 

https://youtu.be/8bglEUfGVrM

**Comment by newenclave** [comment]
@mickrose555 Ah. This video. This video is not about the application. It's about another small project. AppDataGN (https://github.com/newenclave/AtomDataGN). It's a DataField for Garmin devices. DataFields can be used by activities to track and add some extra/custom data to the activity's track. You can add it whenever you want. For example I have Run/Run Atom, Walk/Walk Atom. And the "Atom"s options mean these activities are with AtomFast tracking and the "Dose rate" is stored to Garmin with these.

![image](https://user-images.githubusercontent.com/5781353/103484288-4895c300-4df6-11eb-849c-993b7e012bf2.png)

**Comment by mickrose555** [comment]
my Fenix 6 is on its way from Italy (Ebay 580 Eur) - let me know how to get ready to test...
thanks !

**Comment by newenclave** [comment]
@mickrose555 Congratulation! :) Actually you can install the app from the Garmin Store. But if you have already the f6, I can test multiply connections. So I can build a beta for you. If you have a telegram account I can send it there.  Also I have this repo https://github.com/newenclave/AtomGNBuilds and I can put the build there. Anyway it's quite interesting to see what the Garmin devices can do and what do not.

**Comment by mickrose555** [comment]
> @mickrose555 Congratulation! :) Actually you can install the app from the Garmin Store. But if you have already the f6, I can test multiply connections. So I can build a beta for you. If you have a telegram account I can send it there.  Also I have this repo https://github.com/newenclave/AtomGNBuilds and I can put the build there. Anyway it's quite interesting to see what the Garmin devices can do and what do not. 

oh that would be great!! telegram user: mickrose555
i have 3 more fast coming from Russia so yes, it would be nice to have this passive reading like iOS app ![image](https://user-images.githubusercontent.com/76529467/103930380-ea7f1d80-511e-11eb-884e-658a8039fb4a.jpeg)

**Comment by newenclave** [comment]
@mickrose555 Lets figure out how it can be.

PS: I've written to your telegram.
